### ISSUE TITLE
Multi-GPU support

### ISSUE THREAD

**Issue by InduCherukuri** [TencentARC/ArcNerf#1]
I'm trying to run the training on 4 GPU cores by giving gpu_ids [0,1,2,3] and setting the world_size=4 but the training has been happening only on a single GPU. please let me know what other changes need to be made to run the code on multi-gpu.

Thanks

**Comment by lawy623** [comment]
@InduCherukuri Thanks for using our code!

We support multi-gpu training. But you should not start the training by 
`python train.py --gpu_ids [0,1,2,3] --configs xx`. 

Instead, use 
`python -m torch.distributed.launch --nproc_per_node=4 --master_port=$RANDOM train.py --gpu_ids [0,1,2,3] --configs xx `. 
(See `scripts/gpu.sh` for detail).

I found that there are some bugs in using multi-gpu due to the inconsistency introduced by DistributedDataParallel. I have fixed them in https://github.com/TencentARC/ArcNerf/commit/7fb6a81e1748a23ae2ffde16aab5cc3bb3f7b594. There maybe other problems that I haven't detected, please let me know if you find it.

**Comment by InduCherukuri** [comment]
I have modified the code to run on GPU by modifying the train.py to the following 

def cleanup():
    dist.destroy_process_group()

def training(rank):

    # parse args
    cfgs = parse_configs()

    # trainer
    if valid_key_in_cfgs(cfgs, 'viewer'):
        trainer = ArcNerfNSViewerTrainer(cfgs)
    else:
        trainer = ArcNerfTrainer(rank, cfgs)
    trainer.train()

     cleanup()


if __name__ == '__main__':

    # parse args
    cfgs = parse_configs()
    mp.spawn(training, nprocs=cfgs.dist.world_size)

and changed the default local_rank from 0 to depending on the spawn process

with this I was able to run on multi-gpu

But thanks for your response and the updates you have done to the code. Will check those out, Thank you.

**Comment by InduCherukuri** [comment]
I have couple of question regarding the usage of mask:

1. Is it used to reduce the memory load?
2. How do I bypass this and get the fully rendered image in evaluation instead of masked output?

I have changed mask at ,multiple places and I'm unable to bypass it. I have also changed nerf_model.py to give full output by  adding following line
mask_pts = torch.where(mask_pts==True, False, False)
but I'm getting an error

<img width="1509" alt="error1" src="https://user-images.githubusercontent.com/20127155/227443455-dfc576cc-3c7b-430b-ba4b-ea610b540368.png">
